### PR TITLE
Update README for 1.4

### PR DESCRIPTION
**Pull request by adidahiya**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,38 +205,35 @@ Rule names are always camel-cased and *must* contain the suffix `Rule`. Let us t
 
 Now, let us first write the rule in TypeScript. At the top, we reference TSLint's [definition](https://github.com/palantir/tslint/blob/master/lib/tslint.d.ts) file. The exported class name must always be named `Rule` and extend from `Lint.Rules.AbstractRule`.
 
-```javascript
+```typescript
```
Comment (ashwinr) [inline README.md]: gh markdown supports typescript finally?

Comment (adidahiya) [inline README.md]: it has for a few months I think

File: README.md [hunk 2]
```diff
@@ -205,38 +205,35 @@ Rule names are always camel-cased and *must* contain the suffix `Rule`. Let us t
...
 /// <reference path='tslint.d.ts' />
 
 export class Rule extends Lint.Rules.AbstractRule {
-	public static FAILURE_STRING = "import statement forbidden";
+	  public static FAILURE_STRING = "import statement forbidden";
```
Comment (ashwinr) [inline README.md]: looks like this indentation is off. verify this in the rendered markdown.

File: README.md [hunk 3]
```diff
@@ -257,13 +254,10 @@ function NoImportsWalker() {
...
 NoImportsWalker.prototype.visitImportDeclaration = function (node) {
-    // get the current position and skip over any leading whitespace
-    var position = this.position() + node.leadingTriviaWidth();
-
     // create a failure at the current position
-    this.addFailure(this.createFailure(position, node.width(), "import statement forbidden"));
+    this.addFailure(this.createFailure(node.getStart(), node.width(), "import statement forbidden"));
 
-	// call the base version of this visitor to actually parse this node
+	  // call the base version of this visitor to actually parse this node
```
Comment (ashwinr) [inline README.md]: verify indentation here

File: README.md [hunk 4]
```diff
@@ -283,12 +277,7 @@ Just like rules, additional formatters can also be supplied to TSLint via `--for
 
 export class Formatter extends Lint.Formatters.AbstractFormatter {
     public format(failures: Lint.RuleFailure[]): string {
-        var failuresJSON: any[] = [];
-
-        for (var i = 0; i < failures.length; ++i) {
-            failuresJSON.push(failures[i].toJson());
-        }
-
+        var failuresJSON = failures.map((failure: Lint.RuleFailure) failure.toJson());
```
Comment (ashwinr) [inline README.md]: huh? you're missing a fat arrow

Comment (adidahiya) [inline README.md]: good call

File: README.md (hunk 2 above)
Comment (adidahiya) [inline README.md]: yeah this one is off

File: README.md (hunk 3 above)
Comment (adidahiya) [inline README.md]: got messed up with a mix of tabs & spaces. fixing now

**Pull request merged**